server/journeys/referral/details/tests.test.ts

```diff
@@ -0,0 +1,340 @@
+      .get(`/${uuid}/referral/details`)
+      .expect(200)
+      .expect('Content-Type', /html/)
+
+    const html = createTestHtmlElement(result.text)
+    expect(getByText(html, 'There is a problem')).toBeVisible()
+    const error = getByRole(html, 'link', { name: 'Error message' }) as HTMLLinkElement
+    expect(error).toBeVisible()
+    expect(error.href).toMatch(/#propertyName$/)
+  })
+})
+
+describe('POST /referral/details', () => {
+  it('redirect to /referral/involvement, save incident location, type, date, and time on valid request', async () => {
+    await request(app)
+      .post(`/${uuid}/referral/details`)
+      .type('form')
+      .send({
+        incidentLocation: 'A',
+        incidentType: 'B',
+        hour: '23',
+        minute: '59',
+        incidentDate: '25/12/2009',
+      })
+      .expect(302)
+      .expect('Location', 'involvement')
+
+    expect(reqCaptured.journeyData().referral?.incidentLocation).toEqual({ code: 'A', description: 'TEXT' })
+    expect(reqCaptured.journeyData().referral?.incidentType).toEqual({ code: 'B', description: 'TEXT2' })
+    expect(reqCaptured.journeyData().referral?.incidentDate).toEqual('2009-12-25')
+    expect(reqCaptured.journeyData().referral?.incidentTime).toEqual('23:59')
+  })
+
+  it('redirect to /referral/involvement, save incident location, type, and date (with optional time being empty) on valid request', async () => {
+    await request(app)
+      .post(`/${uuid}/referral/details`)
+      .type('form')
+      .send({
+        incidentLocation: 'A',
+        incidentType: 'B',
+        hour: '',
+        minute: '',
+        incidentDate: '25/12/2009',
+      })
+      .expect(302)
+      .expect('Location', 'involvement')
+
+    expect(reqCaptured.journeyData().referral?.incidentLocation).toEqual({ code: 'A', description: 'TEXT' })
+    expect(reqCaptured.journeyData().referral?.incidentType).toEqual({ code: 'B', description: 'TEXT2' })
+    expect(reqCaptured.journeyData().referral?.incidentDate).toEqual('2009-12-25')
+    expect(reqCaptured.journeyData().referral?.incidentTime).toBeNull()
+  })
+
+  it('redirect to go back and set validation errors if time input is invalid', async () => {
+    await request(app)
+      .post(`/${uuid}/referral/details`)
+      .type('form')
+      .send({
+        incidentLocation: 'A',
+        incidentType: 'B',
+        hour: '12',
+        minute: '',
+        incidentDate: '25/12/2009',
+      })
+      .expect(302)
+      .expect('Location', '/')
+
+    expect(reqCaptured.validationErrors()).toEqual({ incidentTime: ['Enter a time using the 24-hour clock'] })
+  })
+
+  it('redirect to go back and set validation errors if incident date is in the future', async () => {
+    await request(app)
+      .post(`/${uuid}/referral/details`)
+      .type('form')
+      .send({
+        incidentLocation: 'A',
+        incidentType: 'B',
+        hour: '',
+        minute: '',
+        incidentDate: '25/12/2077',
```

Comment:
From a boundary test perspective, it's best to use something like `const tomorrow = addDays(new Date(), 1)` from `date-fns` as that shows the earliest invalid future date